app/controllers/application_controller.rb

```diff
@@ -20,9 +20,57 @@ class ApplicationController < ActionController::Base
 
   def auth_user!
     unless user_signed_in? || devise_controller?
-      session[:previous_url] = request.fullpath unless request.xhr? # do not store AJAX calls
+      session[:previous_url] = request.fullpath unless request.xhr? # do not store AJAX call
       redirect_to new_user_session_path
     end
+    check_path
+  end
+
+  def find_room(id)
+    rooms = current_user.rooms.eager_load(:learning_objects)
+    lo = LearningObject.find(id)
+    room =  rooms.reject{ |m| !(m.learning_objects.include?(lo)) }.first
```

Comment:
Zbytocne sa najskor z DB nacitaju uplne vsetky miestnosti usera, k nim uplne vsetky learning objecty z miestnosti a potom sa to v cykle vo vnutri cyklu (`include` v `reject`) prechadza, aby sa potom vsetko zahodilo nez prvy v poradi a ide nam vlastne iba o IDcko. Na toto sa predsa da spytat rovno databazy: "daj mi vsetky miestnosti, ktore maju tohto usera a tento learning object, zorad podla ID a sprav limit 1"